### PR TITLE
Implement num_integer::Integer for HeaplessBigInt

### PR DESCRIPTION
**Pull request by kaidokert**
The **last Tier-1 parity item** — with this, a `HeaplessBigInt<_, Nct>` is a drop-in `num_traits::PrimInt` **and** `num_integer::Integer`.

Nct-only, mirroring FixedUInt: `gcd`/`lcm`/`div_floor`/`mod_floor`/`divides`/`is_multiple_of`/`is_even`/`is_odd`/`div_rem`.

- **div_floor / mod_floor / div_rem** — the unsigned div/rem (floor == truncating), over the inherent `div_rem`.
- **gcd** — Stein's binary algorithm. Heapless `>>` narrows `len`, so the running values shrink as factors of two are stripped — that's value-correct (comparisons are value-based). **The result is pinned to the operand width `max(len)`: the value is widened before the final `<< shift` so no bit is lost and the width matches `FixedUInt<k>`.** (Straight port of FixedUInt's fixed-width `m << shift` would have narrowed here — a dedicated heapless-internal test covers the sub-capacity case.)
- **lcm** = `a / gcd(a, b) * b`; **is_even/is_odd** — O(1) LSB check like FixedUInt.

**Tests:** generic `for_both_carriers` body (gcd/lcm/div_floor/mod_floor/div_rem/is_multiple_of/parity, incl. coprime and `gcd(0,x)`) plus heapless-internal gcd width-preservation (len 2 in a CAP-8 carrier) and multi-limb gcd/lcm. Verified default / nightly / --no-default-features; clippy clean.

Tier-1 parity is complete. What's left is the optional Tier-3 conveniences (isqrt/ilog/Euclid/abs_diff/Sum-Product/etc.) and the deferred crate-wide signed_shr fix.

## Summary by Sourcery

Implement num_integer::Integer for heapless big integers and verify Tier-1 parity via shared carrier tests.

New Features:
- Support num_integer::Integer for HeaplessBigInt Nct carriers, including gcd, lcm, floor division, modulus, divisibility, parity, and div_rem methods.

Tests:
- Extend NumCarrier tests to cover Integer methods across both carriers, including gcd/lcm edge cases, div_floor/mod_floor/div_rem, divisibility, and parity.
- Add heapless-specific tests to ensure gcd preserves operand width and handles multi-limb values correctly.

### CODE REVIEW

File: src/heapless/num_integer_impl.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        if <Self as Zero>::is_zero(self) && <Self as Zero>::is_zero(other) {
+            return <Self as Zero>::zero();
+        }
+        let gcd = self.gcd(other);
+        *self * (*other / gcd)
+    }
+
+    fn divides(&self, other: &Self) -> bool {
+        self.is_multiple_of(other)
+    }
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/heapless/num_integer_impl.rs]: @gemini-code-assist This one's the reverse — num_integer's own default is `fn divides(&self, other) -> bool { self.is_multiple_of(other) }` (num-integer 0.1.46 lib.rs:206, and it's #[deprecated]). So `divides` == `is_multiple_of` in this crate; the name is a historical misnomer, which is why it's deprecated. My impl matched the default exactly, and `other.is_multiple_of(self)` would actually diverge from the crate and the primitive impls. Took your alternative suggestion and removed the override entirely (1fe1861) so the default handles it. FixedUInt's is likewise not wrong — it matches the same default.

File: tests/carrier_num_traits.rs [hunk 2]
```diff
@@ -196,3 +197,27 @@ fn prim_int() {
...
+        assert_eq!(C::at32(17).gcd(&C::at32(5)), C::at32(1)); // coprime
+        assert_eq!(C::at32(0).gcd(&C::at32(9)), C::at32(9)); // gcd(0, x) = x
+        assert_eq!(C::at32(0).lcm(&C::at32(0)), C::at32(0));
+        // div_floor / mod_floor / div_rem (unsigned: floor == truncating)
+        assert_eq!(C::at32(17).div_floor(&C::at32(5)), C::at32(3));
+        assert_eq!(C::at32(17).mod_floor(&C::at32(5)), C::at32(2));
+        assert_eq!(C::at32(17).div_rem(&C::at32(5)), (C::at32(3), C::at32(2)));
+        // divisibility / parity
+        assert!(C::at32(12).is_multiple_of(&C::at32(4)));
+        assert!(!C::at32(13).is_multiple_of(&C::at32(4)));
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply tests/carrier_num_traits.rs:217]: @gemini-code-assist Skipping these — `divides` is removed now (uses the num_integer default = self.is_multiple_of), and the suggested assertions encode the opposite semantics from the crate's (crate: `4.divides(&12)` = `4.is_multiple_of(12)` = false, not true). The is_multiple_of assertions already cover the underlying predicate.

File: src/heapless/num_integer_impl.rs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                n -= m;
+                n = n >> (PrimBits::trailing_zeros(n) as usize);
+            }
+        }
+        m.widened(width) << (shift as usize)
+    }
+
+    fn lcm(&self, other: &Self) -> Self {
+        if <Self as Zero>::is_zero(self) && <Self as Zero>::is_zero(other) {
+            return <Self as Zero>::zero();
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/heapless/num_integer_impl.rs]: Fixed in 1fe1861 — lcm(0, 0) now returns zero widened to max(self.len(), other.len()) instead of the minimal len-0 zero, so it stays at the operand width. Added a heapless-internal test.

File: src/heapless/num_integer_impl.rs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        let gcd = self.gcd(other);
+        *self * (*other / gcd)
+    }
+
+    fn divides(&self, other: &Self) -> bool {
+        self.is_multiple_of(other)
+    }
+
+    fn is_multiple_of(&self, other: &Self) -> bool {
+        *self % *other == <Self as Zero>::zero()
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/heapless/num_integer_impl.rs:70]: Fixed in 1fe1861 — is_multiple_of now guards `if other.is_zero() { return self.is_zero() }`, matching num_integer's primitive impls, so x.is_multiple_of(&0) is a predicate rather than a % 0 panic. (FixedUInt's is_multiple_of has the same latent panic — worth the same guard on both carriers as a follow-up.)
